scripts/nbd-firewall-config.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+#
+# Copyright (c) Citrix Systems. All rights reserved.
+
+##############################################################
+# Use this script to open/close the specified interfaces
+# (IP addresses) for incoming TCP connections to the NBD port.
+#
+# Usage:
+#   ./nbd-firewall-config set [address...]
+#
+##############################################################
+
+# Historical note: the original version of this file filtered
+# by local IP addresses rather than by network interfaces;
+# that version might one day be a useful reference for something.
+
+# Require the "set" argument because otherwise there would be a risk
+# of someone running the script with no arguments and accidentally
+# setting NBD to be available on no addresses.
+# This way, omitting "set" (or supplying "--help") causes the script
+# to print usage instructions.
+
+set -e
+OP="$1"
+if [ _"${OP}" != _set ]; then
+    echo "Usage: $(basename ${0}) set [interface...]
+will set the firewall to allow incoming TCP connections
+to the NBD port on only the specified interfaces (if any)." 1>&2
+    exit 1
+fi
+shift 1
+
+set -eu
+# No automatic cleanup on error because it would depend too
+# critically on the stage at which the error occurs.
+
+NBDPORT=10809
```

Comment:
Is this port number likely to be changed?

Reply:
Probably not, this is the standard, IANA-reserved NBD port that we SHOULD listen on according to the NBD protocol documentation: https://sourceforge.net/p/nbd/code/ci/master/tree/doc/proto.md